src/nvs.rs

```diff
@@ -110,6 +122,19 @@ impl NvsCustom {
 
         Ok(Self(c_partition))
     }
+
+    fn erase(partition: &str) -> Result<(), EspError> {
+        let mut registrations = NONDEFAULT_LOCKED.lock();
+        let c_partition = CString::new(partition).unwrap();
+
+        if registrations.contains(c_partition.as_ref()) {
+            return Err(EspError::from(ESP_ERR_INVALID_STATE).unwrap());
+        }
```

Comment:
For my understanding: Why is it a invalid state if the registrations contain the partition name?

Maybe also add this as a comment?

Reply:
Comments helped. Thanks :)

Reply:
`esp-idf-svc` doesn't seem to have any proper error reporting for errors originating in it's own code. instead a generic ESP-IDF invalid state error is returned. for what this error means see the now added comments :) good point